### PR TITLE
[BEAM-4602][BEAM-4598] Datetime type comparison

### PR DESCRIPTION
**Pull request by vectorijk**
Support datetime type in comparison expression
Test datetime type comparison at the DSL level

R: @kennknowles @XuMingmin
cc: @akedin @apilloud @amaliujia 

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Spark
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslSqlStdOperatorsTest.java [hunk 1]
```diff
@@ -193,6 +193,29 @@ public void testThatOperatorsExist() {
     }
   }
 
+  @Test
+  @SqlOperatorTest(name = "<", kind = "LESS_THAN")
```
Comment (kennknowles) [inline sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslSqlStdOperatorsTest.java]: Can you leave these lines here, but comment them out? Here is why: when you have an annotation like this it tells the framework that we already have enough tests for this operator. It is great that you added some tests, but we should get a whole suite for the other types too before we say we are done.

Comment (vectorijk) [reply sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslSqlStdOperatorsTest.java]: got it. I have seen #5744 adding tests for comparison operator. I can append my tests in this test case.

**Comment by vectorijk** [comment]
run java precommit

**Comment by vectorijk** [comment]
It's ready to be merged.

**Comment by kennknowles** [comment]
There are three commits but I think it should probably just be one.

**Comment by vectorijk** [comment]
rebased it.

**Pull request merged**